### PR TITLE
Shapella upgrade and scratch deploy

### PR DESCRIPTION
**Pull request by arwer13**
**NB: not yet supposed to be merged or reviewed properly.**

It's supposed to move parallel to shapella-upgrade some time, while the deployment procedure gets refined.

Might get some code related to the upgrade (not-scratch deployment).

### CODE REVIEW

File: scripts/shapella-upgrade/deploy-aragon-implementations.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+const { deployWithoutProxy, deployBehindOssifiableProxy, updateProxyImplementation } = require('../helpers/deploy')
+const { ZERO_ADDRESS, bn } = require('@aragon/contract-helpers-test')
+
+const { APP_NAMES } = require('../constants')
+
+const DEPLOYER = process.env.DEPLOYER || ''
+const REQUIRED_NET_STATE = [
+  `app:${APP_NAMES.LIDO}`,
+  `app:${APP_NAMES.ORACLE}`,
+  `app:${APP_NAMES.NODE_OPERATORS_REGISTRY}`,
```
Comment (TheDZhon) [inline scripts/shapella-upgrade/deploy-aragon-implementations.js:14]: + ELRewardsVault maybe (i.e., the protocol underwent the Merge)

File: scripts/shapella-upgrade/deploy-shapella-contracts.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+echo "NETWORK is $NETWORK"
+
+# yarn install --immutable
+# yarn compile
+
+rm -f deployed-$NETWORK.json
+cp deployed-$NETWORK-defaults.json deployed-$NETWORK.json
+
+yarn hardhat --network $NETWORK run ./scripts/shapella-upgrade/deploy-aragon-implementations.js
+
```
Comment (TheDZhon) [inline scripts/shapella-upgrade/deploy-shapella-contracts.sh:26]: it's not clear who and when deploys non-aragon impls

File: scripts/shapella-upgrade/deploy-non-aragon-contracts-no-proxy-binding.js [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+  `app:${APP_NAMES.ORACLE}`,
+  "app:aragon-agent",
+  "app:aragon-voting",
+  "daoInitialSettings",
+  "oracleReportSanityChecker",
+  "burner",
+  "hashConsensusForAccounting",
+  "hashConsensusForValidatorsExitBus",
+  "withdrawalQueueERC721",
+]
```
Comment (TheDZhon) [inline scripts/shapella-upgrade/deploy-non-aragon-contracts-no-proxy-binding.js:22]: oracle daemon config?

Comment (TheDZhon) [reply scripts/shapella-upgrade/deploy-non-aragon-contracts-no-proxy-binding.js:22]: withdrawal vault

File: scripts/shapella-upgrade/deploy-non-aragon-contracts-no-proxy-binding.js [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+  //
+  // === EIP712StETH ===
+  //
+  await deployWithoutProxy("eip712StETH", "EIP712StETH", deployer, [lidoAddress])
+  logWideSplitter()
+
+  //
+  // === WstETH ===
+  //
+  const wstETHAddress = await deployWithoutProxy("wstETH", "WstETH", deployer, [lidoAddress])
```
Comment (TheDZhon) [inline scripts/shapella-upgrade/deploy-non-aragon-contracts-no-proxy-binding.js:122]: ?

File: scripts/shapella-upgrade/deploy-non-aragon-contracts-no-proxy-binding.js (hunk 3 above)
Comment (arwer13) [reply scripts/shapella-upgrade/deploy-non-aragon-contracts-no-proxy-binding.js:22]: why if we deploy them below and they do not accept any params?

File: scripts/shapella-upgrade/deploy-non-aragon-contracts-no-proxy-binding.js (hunk 4 above)
Comment (arwer13) [reply scripts/shapella-upgrade/deploy-non-aragon-contracts-no-proxy-binding.js:122]: fixed

File: scripts/shapella-upgrade/deploy-shapella-contracts.sh (hunk 2 above)
Comment (arwer13) [reply scripts/shapella-upgrade/deploy-shapella-contracts.sh:26]: these instructions belong to the deploy plan, don't they?

File: scripts/shapella-upgrade/deploy-aragon-implementations.js (hunk 1 above)
Comment (arwer13) [reply scripts/shapella-upgrade/deploy-aragon-implementations.js:14]: okay

**Pull request merged**